src/Microsoft.Health.Dicom.SqlServer/Features/ExtendedQueryTag/SqlExtendedQueryTagStoreV3.cs

```diff
@@ -19,5 +24,16 @@ internal class SqlExtendedQueryTagStoreV3 : SqlExtendedQueryTagStoreV2
         }
 
         public override SchemaVersion Version => SchemaVersion.V3;
+        public override async Task<IReadOnlyList<ExtendedQueryTagStoreEntry>> GetExtendedQueryTagsAsync(int limit, int offset, CancellationToken cancellationToken = default)
+        {
+            var tags = await GetAllExtendedQueryTagsAsync(cancellationToken);
+            tags.Sort((entry1, entry2) => entry1.Key - entry2.Key);
+            if (offset < 0 || offset >= tags.Count || limit <= 0)
```

Comment:
nit: let's check with `EnsureArgs` for `offset < 0` and `limit <= 0`, as those are invalid.